### PR TITLE
Add BCD for CSS Anchor Positioning spec

### PR DESCRIPTION
**Pull request by chrisdavidmills**
#### Summary

[CSS Anchor Positioning](https://www.w3.org/TR/css-anchor-position-1/) is set to be released in Chrome 125. The associated [Chrome Status entry](https://chromestatus.com/feature/5124922471874560) does not appear to be very up-to-date, but I have been assured by Chrome engineering that this is the case.

This PR adds BCD for all the features in the spec, namely:

- The `anchor-name` property
- The `anchor-scope` property
- The `position-anchor` property
- The `position-visibility` property
- The `anchor()` function
- I've also added a subfeature to each of the properties that support `anchor()` as a value:
  - `top`
  - `left`
  - `bottom`
  - `right`
  - `inset-block-start`
  - `inset-block-end`
  - `inset-inline-start`
  - `inset-inline-end`
  - `inset-block`
  - `inset-inline`
  - `inset`
- The `anchor-center` value, which is supported on the following properties (I've added a subfeature for each):
  - `justify-self`
  - `align-self`
  - `justify-items`
  - `align-items`
- The `anchor-size()` function
- I've also added a subfeature to each of the properties that support `anchor-size()` as a value:
  - `width`
  - `height`
  - `min-width`
  - `min-height`
  - `max-width`
  - `max-height`
  - `block-size`
  - `inline-size`
  - `min-block-size`
  - `min-inline-size`
  - `max-block-size`
  - `max-inline-size`
- The `inset-area` property
- The `position-try-options` property
- The `position-try-order` property
- The `position-try` shorthand property
- The `@position-try` at-rule
- The `CSSPositionTryRule` CSSOM interface and its properties:
  - `name`
  - `style` 

Note that there are features that don't appear to be supported yet in Chrome, as of the creation of this PR — the `anchor-scope` property, and the `position-try*` properties/`@position-try` at-rule. AFAIUI, these should be added/enabled fairly soon, so I've added them in for now. I'll hopefully get a confirmation soon either way, so I can remove/retest as appropriate.

Note that there are also a couple of features that Chrome supports but aren't in a spec yet (see https://github.com/whatwg/html/pull/9144/files):

- The HTML `anchor` attribute, used to declare the ID of an element that you want this element to be anchored to (i.e. its anchor element).
- The `Element.anchorElement` property, used to access a reference to an element's anchor element.

I also added in data points for Chrome/CxA 125, so that my BCD doesn't error out.

### CODE REVIEW

**Comment by lukewarlow** [comment]
I'm very dubious of that version 125 number. There's not afaik been an intent to ship yet, and https://chromiumdash.appspot.com/schedule suggests the cut off for Chrome 125 was April 1st?

**Comment by chrisdavidmills** [comment]
> I'm very dubious of that version 125 number. There's not afaik been an intent to ship yet, and https://chromiumdash.appspot.com/schedule suggests the cut off for Chrome 125 was April 1st?

@lukewarlow yeah, that's a good point. @mfreed7 / @rachelandrew, do you have any updated information on this?

**Comment by mfreed7** [comment]
> > I'm very dubious of that version 125 number. There's not afaik been an intent to ship yet, and https://chromiumdash.appspot.com/schedule suggests the cut off for Chrome 125 was April 1st?
> 
> @lukewarlow yeah, that's a good point. @mfreed7 / @rachelandrew, do you have any updated information on this?

https://groups.google.com/a/chromium.org/g/blink-dev/c/jGTYNuidPRs

File: api/Element.json
Comment (Elchi3) [inline api/Element.json:226]: The spec PR adds this to `HTMLElement` not `Element`. Also, I don't see Chrome 125 shipping this. Should we remove this for now?

Comment (lukewarlow) [reply api/Element.json:226]: Yeah this is not included in what's shipping so I wouldn't include it in this dataset.

File: css/properties/inset-area.json
Comment (Elchi3) [inline css/properties/inset-area.json:4]: This one is missing a lot of values as sub features. Running the collector I got this: https://gist.github.com/Elchi3/7651d3f1bcfbe328645165866120010a. Do you want to add it?

File: css/properties/position-try-options.json
Comment (Elchi3) [inline css/properties/position-try-options.json:4]: Missing sub features: flip-block, flip-inline, flip-start, none.

File: css/properties/position-try-order.json
Comment (Elchi3) [inline css/properties/position-try-order.json:4]: Missing sub features: most-block-size, most-height, most-inline-size, most-width, normal

File: css/properties/position-visibility.json
Comment (Elchi3) [inline css/properties/position-visibility.json:4]: Missing sub features: always  anchors-valid, anchors-visible, no-overflow

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Thanks for your work on this Chris. I double-checked all of this against the @openwebdocs BCD collector and left some comments.
- need to have sub features for property values.
- anchorElement should probably be removed
- anchor-scope seem unsupported, also remove?
- What about adding the CSSPositionTryDescriptors interface?

File: api/Element.json
Comment (chrisdavidmills) [reply api/Element.json:226]: Ooops! In my next commit, I've moved it to `HTMLElement`.

The thing is though, it is supported, not behind a flag, in Canary (I've tested it). I'm guessing it just won't be shipped in Chrome stable for some time. So I feel like it is useful to keep this data for folks doing testing of preview features.

How can we represent features in this state? I thought we used to have some kind of "preview" value to use for this, but I can't find mention of it in the BCD data schema doc.

File: css/properties/anchor-scope.json
Comment (chrisdavidmills) [reply css/properties/anchor-scope.json:1]: Ah yes, this isn't supported yet. Removed in next commit.

File: css/properties/position-visibility.json
Comment (chrisdavidmills) [reply css/properties/position-visibility.json:4]: added in next commit

File: css/properties/position-try-order.json
Comment (chrisdavidmills) [reply css/properties/position-try-order.json:4]: Value subfeatures added

File: css/properties/position-try-options.json
Comment (chrisdavidmills) [reply css/properties/position-try-options.json:4]: I've added the list value subfeatures. However, this property can also take an `inset-area` value as a value, to use as a position try option, and it can take a dashed ident as an option, which represents a custom position try option defined inside a `@position-try` at-rule. Do we not need separate sub-features for these value options?

File: css/properties/inset-area.json
Comment (chrisdavidmills) [reply css/properties/inset-area.json:4]: Sure, added.

One thing to flag is that these are not exclusively single, standalone values. They can be used on their own, but they can also be used in combination, as pairs of values. Is that an issue? I will of course be explaining all of this in the reference page.

**Comment by chrisdavidmills** [comment]
@Elchi3 ok, I think I've done everything you asked for here, except for `CSSPositionTryDescriptors` — that wasn't in the spec when I first submitted this PR.

Can you show me how to go about generating BCD for this? There's over 70 properties 😱

**Comment by Elchi3** [comment]
Thanks Chris, so now you know why I said let's generate this stuff :))

A gist for you for CSSPositionTryDescriptors from the collector: https://gist.github.com/Elchi3/2fcdc552ec5defba79ffd789e398f901 (Doesn't yet contain spec_urls.)

File: api/HTMLElement.json
Comment (Elchi3) [inline api/HTMLElement.json:124]:
```suggestion
              "version_added": "preview",
              "partial_implementation": true,
              "notes": "Supported on <code>Element</code>, not on <code>HTMLElement</code>"
```

File: api/Element.json
Comment (Elchi3) [reply api/Element.json]: Thanks for moving it. See https://github.com/mdn/browser-compat-data/pull/22813#pullrequestreview-2011180144 for setting preview and partial implementation because it is on the wrong interface at the moment.

File: css/properties/inset-area.json
Comment (Elchi3) [reply css/properties/inset-area.json:4]: That's okay, if there are compat issues with the values standalone or in combination, we will see how to add it in the future. For now it is important to capture which values are part of the initial implementation.

It would be nice if you add spec_urls (this is our insurance that the values are valid per spec and not outdated)

File: css/properties/position-try-options.json
Comment (Elchi3) [reply css/properties/position-try-options.json:4]: I think it would be great to add them in but I'm not going to block this PR if they aren't here for now.

File: api/HTMLElement.json
Comment (chrisdavidmills) [reply api/HTMLElement.json]: Great! Updated as advised.

File: api/Element.json
Comment (mfreed7) [reply api/Element.json]: Just to confirm - the `anchor` attribute won’t ship in 125. (And current Canaries and Beta should not support it - Chris, please file a bug if you see some observable support.) We are still hoping to ship it soon once the spec discussion concludes, but I don’t know when that will be.

File: css/properties/anchor-scope.json
Comment (mfreed7) [reply css/properties/anchor-scope.json:1]: Chrome plans to ship this property ASAP, hopefully in 126. Just FYI.

File: css/properties/inset-area.json
Comment (chrisdavidmills) [reply css/properties/inset-area.json:4]: spec urls added!

I'll add spec urls for the other property subfeatures I added too.

File: api/Element.json
Comment (chrisdavidmills) [reply api/Element.json]: @mfreed7 OK, filed at https://issues.chromium.org/issues/335744345. Is that OK? I've never filed a Chromium bug before ;-)

So I should just remove this for now?

And, can I clarify — this is just the `anchorElement` property we are talking about? The anchor` HTML attribute is still shipping?

File: css/properties/position-visibility.json
Comment (chrisdavidmills) [reply css/properties/position-visibility.json:4]: OK, got it. I've removed it from the data for now.

File: css/properties/anchor-scope.json
Comment (chrisdavidmills) [reply css/properties/anchor-scope.json:1]: Cool, thanks @mfreed7; I'll note that in the main research doc. I can add it as a follow-up work item if needs be.

**Review by Elchi3 — APPROVED** [review]
Thanks Chris! Ready from my side! 👍

**Comment by chrisdavidmills** [comment]
Super awesome, thanks @Elchi3!

File: api/Element.json
Comment (foolip) [reply api/Element.json]: I'm removing the `anchor` attribute in https://github.com/mdn/browser-compat-data/pull/22910.

Comment (mfreed7) [reply api/Element.json]: Thanks for filing the bug - I just commented there. I suspect you have Experimental Web Platform Features enabled, which is why you see the `anchorElement` property available. Neither the `anchor` content attribute nor the `anchorElement` IDL attribute are shipping in Chrome in 125 or likely 126 at least.

Comment (mfreed7) [reply api/Element.json]: > @mfreed7 OK, filed at https://issues.chromium.org/issues/335744345. Is that OK? I've never filed a Chromium bug before ;-)

Also, yes of course, that was perfect!

**Pull request merged**
